functions/src/sendEmail.js

```diff
@@ -0,0 +1,37 @@
+
+
+exports.sendEmailOnNewAllotment = (snapshot, db, request, sendInBlueSecretKey, sendEmail) => {
+
+    const original = snapshot.val();
+    let files_updated = true;
+
+    if (original.files && original.list)
+        original.files.forEach(file => {
+            ref = db.ref(`/sqr/files/${original.list}/${file}`);
+
+            // check if the file exists first
+            ref.child("status").once('value')
+            .then(snapshot => {
+                if(snapshot.exists())
+                    ref.update({status: "Given"}, err => {
```

Comment:
You're updating the status of a file inside the `sendEmailOnNewAllotment` function. Not a good practice, as the method name does not reveal all the side-effects. As I mentioned in Trello, it may worth creating two functions for two actions, not to mix things together.

Reply:
done